### ISSUE TITLE
Purchase date is sometimes download date?

### ISSUE THREAD

**Issue by avwohl** [openaudible/openaudible#1441]
I have 1612 books. I hadn't synced in a couple of months, so I got the current open-audible. After the sync, about 25 books had a purchase date of 8-1-2024, but they were all purchased earlier.

**Comment by openaudible** [comment]
If we can get the purchase date from your Audible account we use that. But if you drag and drop or import an audiobook from another source we won’t know it so we use the import date. 
You could try to use the get info and use the auto update and if the file is in your library it might reset the date correctly.

**Comment by openaudible** [comment]
A full rescan should also set the date correctly. Will look again but it seems to be working on our end.

**Comment by avwohl** [comment]
A full rescan did not fix it.  I did fix it. First, if you do a full sync with multiple accounts and one of them and grown a red error indicator on the account it should say the sync wasn't really full as one account could not be accessed.

Next after fixing that (deleted red account, re-added and did a full sync) it still would not fix the date on the ~15 files.  Then I noticed that the 15 files had file name and purchase date (which was download date) only, no other fields.  I deleted all 15, did a sync then it was happy all data ok.

The 15 files with missing data didn't get fixed by full sync yet they blocked download of a good copy.  It seems to me if a file can block a download all the fields should be fixed or verified on full sync.
